google-cloud-core/src/main/java/com/google/cloud/BillingName.java

```diff
@@ -0,0 +1,168 @@
+
+import com.google.api.pathtemplate.PathTemplate;
+import com.google.api.resourcenames.ResourceNameType;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+// AUTO-GENERATED DOCUMENTATION AND CLASS
+@javax.annotation.Generated("by GAPIC protoc plugin")
+public class BillingName extends ParentName {
+
+  private static final PathTemplate PATH_TEMPLATE =
+      PathTemplate.createWithoutUrlEncoding("billingAccounts/{billing_account}");
+
+  private volatile Map<String, String> fieldValuesMap;
+
+  private final String billingAccount;
+
+  public String getBillingAccount() {
+    return billingAccount;
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  private BillingName(Builder builder) {
+    billingAccount = Preconditions.checkNotNull(builder.getBillingAccount());
+  }
+
+  public static BillingName of(String billingAccount) {
+    return newBuilder().setBillingAccount(billingAccount).build();
+  }
+
+  public static String format(String billingAccount) {
+    return newBuilder().setBillingAccount(billingAccount).build().toString();
+  }
+
+  public static BillingName parse(String formattedString) {
+    if (formattedString.isEmpty()) {
+      return null;
+    }
+    Map<String, String> matchMap =
+        PATH_TEMPLATE.validatedMatch(
+            formattedString, "BillingName.parse: formattedString not in valid format");
+    return of(matchMap.get("billing_account"));
+  }
+
+  public static List<BillingName> parseList(List<String> formattedStrings) {
+    List<BillingName> list = new ArrayList<>(formattedStrings.size());
+    for (String formattedString : formattedStrings) {
+      list.add(parse(formattedString));
+    }
+    return list;
+  }
+
+  public static List<String> toStringList(List<BillingName> values) {
+    List<String> list = new ArrayList<String>(values.size());
+    for (BillingName value : values) {
+      if (value == null) {
+        list.add("");
+      } else {
+        list.add(value.toString());
+      }
+    }
+    return list;
+  }
+
+  public static boolean isParsableFrom(String formattedString) {
+    return PATH_TEMPLATE.matches(formattedString);
+  }
+
+  public Map<String, String> getFieldValuesMap() {
+    if (fieldValuesMap == null) {
+      synchronized (this) {
+        if (fieldValuesMap == null) {
+          ImmutableMap.Builder<String, String> fieldMapBuilder = ImmutableMap.builder();
+          fieldMapBuilder.put("billingAccount", billingAccount);
+          fieldValuesMap = fieldMapBuilder.build();
+        }
+      }
+    }
+    return fieldValuesMap;
+  }
+
+  public String getFieldValue(String fieldName) {
+    return getFieldValuesMap().get(fieldName);
+  }
+
+  /** @deprecated This method is only present to satisfy the ResourceName interface. */
+  @Deprecated
```

Comment:
This can be removed now (we'll remove it from what the protoc plugin generates later too)

Reply:
Done